apps/cli/src/exec/mksquashfs.ts

```diff
@@ -0,0 +1,77 @@
+import { parse, SemVer } from "semver";
+import { execaDockerFallback } from "./util";
+
+export const fromDirectory = (options: {
+    cwd?: string;
+    image?: string;
+    input: string;
+    output: string;
+}) => {
+    const { cwd, image, input, output } = options;
+    const compression = "lzo"; // make customizable? default is gzip
+    const command = "mksquashfs";
+    const args = [
+        input,
+        output,
+        "-all-time",
+        "0",
+        "-all-root", // XXX: should we use this?
+        "-noappend",
+        "-comp",
+        compression,
+        "-no-progress",
+    ];
+    return execaDockerFallback(command, args, { cwd, image });
+};
+
+export const fromTar = (options: {
+    cwd?: string;
+    image?: string;
+    input: string;
+    output: string;
+}) => {
+    const { cwd, image, input, output } = options;
+    const compression = "lzo"; // make customizable? default is gzip
+    const command = "mksquashfs";
+    const args = [
+        "-",
+        output,
+        "-tar",
+        "-all-time",
+        "0",
+        "-all-root", // XXX: should we use this?
+        "-noappend",
+        "-comp",
+        compression,
+        "-no-progress",
```

Comment:
Same as previous comment for `genext2fs` module.

Reply:
done